### PR TITLE
Add JavaScript for linear search

### PR DESCRIPTION
**Pull request by JoseHung**
### Checklist

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

I also fixed a error of making assignment to the constant variable in the ListNode.js.

### CODE REVIEW

File: codes/javascript/chapter_searching/binary_search.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+/* 二分查找（左闭右开） */
+function binarySearch1(nums, target) {
+    // 初始化左闭右开 [0, n) ，即 i, j 分别指向数组首元素、尾元素+1
+    let i = 0, j = nums.length;
+    // 循环，当搜索区间为空时跳出（当 i = j 时为空）
+    while (i < j) {
+        let m = parseInt((i + j) / 2);    // 计算中点索引 m ，在 JS 中需使用 parseInt 函数取整
+        if (nums[m] < target)             // 此情况说明 target 在区间 [m+1, j) 中
+            i = m + 1;
+        else if (nums[m] > target)        // 此情况说明 target 在区间 [i, m-1] 中
```
Comment (krahets) [inline codes/javascript/chapter_searching/binary_search.js:34]: [i, m)

File: docs/chapter_searching/binary_search.md [hunk 2]
```diff
@@ -337,7 +369,10 @@ $$
 === "JavaScript"
 
     ```js title=""
-
+    // (i + j) 有可能超出 int 的取值范围
+    let m = parseInt((i + j) / 2);
+    // 更换为此写法则不会越界
+    let m = parseInt(i + (j - i) / 2);
```
Comment (krahets) [inline docs/chapter_searching/binary_search.md:375]: The max value of the JS number is 1.7976931348623157e+308. Therefore, I don't think we have to deal with big-number problem like Java.

**Review by krahets — COMMENTED** [review]
Thank @JoseHung for the addition and the fixing! There are several comments to be addressed.

File: docs/chapter_searching/binary_search.md [hunk 3]
```diff
@@ -337,7 +369,8 @@ $$
 === "JavaScript"
 
     ```js title=""
-
+    // JavaScript 中的数字足够大（最大数值为 1.7976931348623157e+308 ）
```
Comment (krahets) [inline docs/chapter_searching/binary_search.md:372]: My bad, 1.7976931348623157e+308 is not infinite, so we still need to handle the big-number problem. I'll update the code.

**Review by krahets — APPROVED** [review]
Thanks for the update! Now merge it.

File: docs/chapter_searching/binary_search.md (hunk 3 above)
Comment (JoseHung) [reply docs/chapter_searching/binary_search.md]: Hahaha. Actually I think both of the explanations are reasonable.

**Pull request merged**